### PR TITLE
Stirling/challenge change

### PR DESCRIPTION
**Pull request by stirlingwaite**
**Pull Request Summary:**
Updated the DS code challenge to be more ML Engineer focused.

1. Added the ML notebook from Amir
2. Re-factored software engineer API code challenge to work for our use case
3. Updated README.md to reflect those changes
4. Condensed README.md to just one readme, instead of having 3 different readme's.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,108 @@
...
+
+### Expectations
+You can expect this test to use the following technologies:
+
+- PostGres
+- Python
+- Jupyter Notebook
+- Docker/Docker-compose
+- Flask (API)
+- PyTorch (classifier, Bart-Base)
```
Comment (AmirAktify) [inline README.md:15]: I ended up switching `bart-base` out for `distilbert` in the notebook, can we modify this accordingly please?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,108 @@
...
-
-- [Getting started](#getting-started)
-- [Docker Compose](#docker-compose)
-- [Architecture](#architecture)
-    - [Database](#database)
-    - [API](#api)
-
+## Goal
+
+>The goal of this test is not ment to be exhaustive, but we do want to test your skills in analytical thinking, architecture patterns, data engineering and machine learning engineering.
```
Comment (AmirAktify) [inline README.md:5]: "not ment to be" => "not meant to be"

File: README.md [hunk 3]
```diff
@@ -41,19 +131,68 @@ When we started the development environment we used the `-d` option. Use of the
...
+
+### API Patterns
+
+The API makes use of a couple of Flask patterns for decomposition and testing.
+
+The first is the [Blueprint Pattern](https://flask.palletsprojects.com/en/1.1.x/blueprints/). Blueprints are found under [services/api/src/blueprints](services/api/src/blueprints). Blueprints allow us to break down our code base into smaller, related chunks.
+
+The second pattern is the [Application Factory Pattern](https://flask.palletsprojects.com/en/1.1.x/patterns/appfactories/). The factory allows us to provide configuration to our application on demand and offers a cleaner bootstrapping process (see [services/api/src/__init__.py](services/api/src/__init__.py)) and greater control over configuration during unit testing.
+
+The final pattern is the [Service Layer Pattern](https://www.oreilly.com/library/view/architecture-patterns-with/9781492052197/ch04.html). This pattern enables co-reuse and greater control over automated testing.
```
Comment (AmirAktify) [inline README.md:165]: "enables co-reuse" => "enables code-reuse"

**Review by AmirAktify — COMMENTED** [review]
Just a few minor typos I noted, otherwise looks good to me.

File: README.md (hunk 3 above)
Comment (stirlingwaite) [reply README.md]: Made the suggested change

File: README.md (hunk 2 above)
Comment (stirlingwaite) [reply README.md]: Made the suggested change

File: README.md (hunk 1 above)
Comment (stirlingwaite) [reply README.md]: Made the suggested change

**Pull request merged**